libraries/core_libs/consensus/src/final_chain/final_chain.cpp

```diff
@@ -316,15 +338,30 @@ class FinalChainImpl final : public FinalChain {
     blk_header.extra_data = extra_data;
     dev::BytesMap trxs_trie, receipts_trie;
     dev::RLPStream rlp_strm;
-    for (size_t i(0); i < transactions.size(); ++i) {
-      auto const& trx = transactions[i];
-      auto i_rlp = util::rlp_enc(rlp_strm, i);
+    auto receipt_idx = 0;
```

Comment:
optional: transation_id probably makes more sense as you are itereting over trasactions and system_transaction - receipts then depend on transaction and its id...